architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+
+app-config.local.yaml has settings for dev environment and merges and overlays the configs in app-config.yaml. 
+app-config.dashboard.yaml has settings for dashboard installation as part of Radius installation. This again merges and overlays the configs in app-config.yaml. 
+
+We have used these files to configure our application to
+
+1. allow communication using Radius API. The data for rendering Radius visuals is obtained by calling different Radius APIs. 
+
+```
+app-config.local.yaml:
+backend:
+  # Allow the backend to make requests to the local Kubernetes cluster.
+  reading:
+    allow:
+      - host: localhost:8001
+      - host: 127.0.0.1:8001
+```
+```
+app-config.dashboard.yaml
+
+kubernetes:
+  serviceLocatorMethod:
+    type: singleTenant
+  # Use the local proxy on localhost:8001 to talk to the local Kubernetes cluster.
+  clusterLocatorMethods:
+    - type: config
+      clusters:
+        - name: self
+          # The URL to the in-cluster Kubernetes API server.
+          # Backstage docs state it should be ignored when in-cluster, but it appears to be used.
+          url: https://kubernetes.default.svc.cluster.local
+```
+
+2. not expose backend directly.
+
+```
+app-config.yaml
+
+backend:
+  # Note that the baseUrl should be the URL that the browser and other clients
+  # should use when communicating with the backend, i.e. it needs to be
+  # reachable not just from within the backend host, but from all of your
+  # callers. When its value is "http://localhost:7007", it's strictly private
+  # and can't be reached by others.
+  baseUrl: http://localhost:7007
+  # The listener can also be expressed as a single <host>:<port> string. In this case we bind to
+  # all interfaces, the most permissive setting. The right value depends on your specific deployment.
+  listen: ':7007'
+```
+
+1.  use sqlite DB. This is light weight, not accessible to users directly and contains no useful information.
+  
+```
+app-config.yaml :
+
+backend:
+  database:
+    client: better-sqlite3
+    connection: ':memory:'
+```
+
+At present, Dashboard can only present the Radius application metadata visually. It has no ability to Create, Modify, Update or Delete any of the Radius application resources. This eliminates the scope of threats that require "write" action using the Dashboard portal.
+
+#### Storage of secrets
+
+None
+
+#### Data Serialization / Formats
+
+None
+
+### Cryptography 
+
+None
+
+### Clients
+
+The primary user of Dashboard is browser. At present, we dont have any other Backstage plugin that cloud be a Radius Dashboard client but that could change in future. 
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Dashboard component:
+
+- **Kubernetes Cluster**: The overall environment where the Dashboard component operates and serves clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+#### Key Points of Namespaces
```

Comment:
Do we need to copy all of these details from the Controller document?

Reply:
What would happen if we didn't?

Reply:
Add details on how namespace is helpful in RBAC